### PR TITLE
@mzikherman: Track more pages' load times #migration

### PR DESCRIPTION
**Pull request by joeyAghion**
We currently record "events" corresponding to end-users' page-load times to both Segment and Datadog (via [Volley](https://github.com/artsy/volley)), but only for the home page, artist page, and collect page.

This PR does a few things:
* Stops sending page-load events to Segment, where they risk exhausting our expensive quota.
* Respects a new `sd.PAGE_TYPE` setting when labeling the event metrics still sent to Datadog. This is necessary if we want to track pages whose paths can't be used to determine a label (like partner profiles, fair microsites, etc.).
* Minor: adds a `.nvmrc` file.

Most of the work went into finding the best/simplest location to set the `sd.PAGE_TYPE` value for pages or whole sets of related pages. This wasn't easy and @acjay and I wished there was a more straightforward way to align this with the declared routes explicitly, but we found that set-up code from some routes was executed as part of later routes in the sequence. (E.g., route handlers in `profile`, `partner`, and `partner2` app directories are all evaluated for any fair-related pages.)

**Post-deploy migration**

    heroku config:set --app force-production TRACK_PAGELOAD_PATHS=home|collect|artist|artwork|show|article|feature|gene|tag|partner|fair # or similar

### CODE REVIEW

**Comment by acjay** [comment]
For the migration, we'd want `home`, instead of the empty string, right?

**Comment by joeyAghion** [comment]
Yup--you're right. Added `home` as well as `partner` and `fair` to the migration.

Also pushed a spec fix.

File: src/desktop/analytics/main_layout.js [hunk 1]
```diff
@@ -19,33 +19,25 @@ if (
...
-    _.each(sd.TRACK_PAGELOAD_PATHS.split('|'), path => {
-      if (topLevelPath === path) {
-        window.setTimeout(function() {
-          const {
-            requestStart,
-            loadEventEnd,
-            domComplete,
-          } = window.performance.timing
+    const pageType =
+      window.sd.PAGE_TYPE || window.location.pathname.split('/')[1]
```
Comment (mzikherman) [inline src/desktop/analytics/main_layout.js]: looks like you can just use `sd.`, and not prefix it with `window` #minor

File: src/desktop/apps/partner/routes.coffee [hunk 2]
```diff
@@ -1,53 +1,47 @@
+@requirePartner = (req, res, next) ->
+  return next('route') unless res.locals.profile?.isPartner()
```
Comment (mzikherman) [inline src/desktop/apps/partner/routes.coffee]: 👍

**Comment by mzikherman** [comment]
Looks great! A PR from @joeyAghion in Force!! 


tenor

**Comment by joeyAghion** [comment]
Migration applied in staging and production!

**Pull request merged**